### DISCUSSION TITLE
chatgpt没用明白

### DISCUSSION THREAD

**Discussion by Miuzarte** [Tsuk1ko/CQ-picfinder-robot#382] [问题求助] [answered]
怎么发都没反应，日志也没有
api是从[https://beta.openai.com/account/api-keys](https://beta.openai.com/account/api-keys)里拿的吧，看了眼18刀都在
![R C5XB%MDUSLVG(MTK Y9G5](https://user-images.githubusercontent.com/66856838/208168368-17d82557-84ce-43c0-8dc3-f5aa58ef8014.png)
![image](https://user-images.githubusercontent.com/66856838/208168872-488212bf-54dc-42e3-9fef-467683406da0.png)

**Comment by Tsuk1ko** [comment]
是，开 `bot.debug` 看看有日志不

**Comment by Tsuk1ko** [comment]
草，突然发现黑白名单功能没做（

虽然不影响主要功能

**Comment by Miuzarte** [comment, marked as answer]
好像还是没东西，黑白名单清空了
```log
2022/12/17 02:58:53 配置已重载
2022/12/17 02:58:53 收到私聊消息 qq=
--reload
2022/12/17 02:59:04 收到群组消息 group=qq=
chatgpt你是谁
2022/12/17 02:59:20 配置已重载
2022/12/17 02:59:20 收到私聊消息 qq=982809597
--reload
2022/12/17 02:59:27 收到群组消息 group=qq=
chatgpt你是谁
2022/12/17 02:59:51 收到群组消息 group=qq=
/help
2022/12/17 02:59:51 回复群组消息 group=qq=
搜图帮助：
私聊 - 直接发送[图片]{参数}
群聊 - 大喊“我要搜图”|“/wyst”再发送[图片]{参数}
群聊 - 回复那张[图片]并@BOT{参数}
群聊 - @BOT同时发送[图片]{参数}
参数：
--get-url：获取图片的在线链接（不会搜图）
--purge：无视缓存搜图
--all：全库搜索（默认）
--pixiv：从P站中搜索
--danbooru：从 Danbooru 中搜索
--doujin/book：搜索本子
--anime：搜索番剧
--a2d：使用 ascii2d 进行搜索（优势搜索局部图能力较强）
--akhr/公招：明日方舟公开招募计算
听说同时发送主人的昵称、QID和QQ号会发生神秘的事情
[CQ:image,file=../sthelp_v3.gif]
```

**Reply by Tsuk1ko** [reply]
那啥，不会是根本没更新代码吧

**Reply by Miuzarte** [reply]
> 那啥，不会是根本没更新代码吧

![image](https://user-images.githubusercontent.com/66856838/208170495-926d6e2e-d4be-4ce4-98df-a838191e5150.png)
我每次更新都是解压覆盖

**Reply by Tsuk1ko** [reply]
你要不试试把除了 `data` `node_modules` `config.jsonc` 之外的都删了再解压一遍

**Reply by Tsuk1ko** [reply]
啊，难道说你跑的还是旧版本，因为 pm2 配置没更新

**Reply by Tsuk1ko** [reply]
> 你要不试试把除了 `data` `node_modules` `config.jsonc` 之外的都删了再解压一遍

这个做完以后

```
npm run kill
npm i
npm start
```

**Reply by Miuzarte** [reply]
> > 你要不试试把除了 `data` `node_modules` `config.jsonc` 之外的都删了再解压一遍
> 
> 这个做完以后
> 
> ```
> npm run kill
> npm i
> npm start
> ```

解决了，是以后每次更新都要npm run kill一遍吗

**Reply by Tsuk1ko** [reply]
不是，是因为 3.0.0 开始全改成原生 esm 了，文件扩展名全变成 `.mjs`，所以你直接解压实际上一点都没替换到，pm2 又没更新配置，所以其实一直启动的还是 2.x 版本

https://github.com/Tsuk1ko/cq-picsearcher-bot/blob/557a3d64a76185e4f8d1d7bedbb58e8f7cff2727/CHANGELOG.md?plain=1#L21-L36
